feature-push-notifications/src/main/java/io/novafoundation/nova/feature_push_notifications/data/presentation/welcome/PushWelcomeViewModel.kt

```diff
@@ -0,0 +1,79 @@
+package io.novafoundation.nova.feature_push_notifications.data.presentation.welcome
+
+import android.Manifest
+import android.os.Build
+import androidx.lifecycle.MutableLiveData
+import io.novafoundation.nova.common.base.BaseViewModel
+import io.novafoundation.nova.common.mixin.api.Retriable
+import io.novafoundation.nova.common.mixin.api.RetryPayload
+import io.novafoundation.nova.common.resources.ResourceManager
+import io.novafoundation.nova.common.utils.Event
+import io.novafoundation.nova.common.utils.permissions.PermissionsAsker
+import io.novafoundation.nova.common.view.ButtonState
+import io.novafoundation.nova.common.view.dialog.retryDialog
+import io.novafoundation.nova.feature_push_notifications.R
+import io.novafoundation.nova.feature_push_notifications.data.PushNotificationsRouter
+import io.novafoundation.nova.feature_push_notifications.data.domain.interactor.PushNotificationsInteractor
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+
+class PushWelcomeViewModel(
+    private val router: PushNotificationsRouter,
+    private val pushNotificationsInteractor: PushNotificationsInteractor,
+    private val permissionsAsker: PermissionsAsker.Presentation,
+    private val resourceManager: ResourceManager
+) : BaseViewModel(), PermissionsAsker by permissionsAsker, Retriable {
+
+    private val _enablingInProgress = MutableStateFlow(false)
+
+    override val retryEvent: MutableLiveData<Event<RetryPayload>> = MutableLiveData()
+
+    val buttonState = _enablingInProgress.map { inProgress ->
+        when (inProgress) {
+            true -> ButtonState.PROGRESS
+            false -> ButtonState.NORMAL
+        }
+    }
+
+    fun backClicked() {
+        router.back()
+    }
+
+    fun termsClicked() {
+        // Need to implement
+    }
+
+    fun privacyClicked() {
+        // Need to implement
```

Comment:
Are there any pending things stopping us to implement it? Iirc those links are present in AppLinksProvider

Reply:
Yes, as I know we are going to create special pages for push terms and privacy